app/src/components/RegisterArtifact.tsx

```diff
@@ -68,13 +68,57 @@ interface ArtifactMetadata {
   documents: ArtifactDocument[];
 }
 
+interface Artist {
+  id: number;
+  name: string;
+  wallet: string;
+  nationality: string;
+  birthYear: string;
+  deathYear: string;
+}
+
 const RegisterArtifact: React.FC = () => {
+  const [artist, setArtist] = React.useState<Artist>({
+    id: 0,
+    name: '',
+    wallet: '',
+    nationality: '',
+    birthYear: '',
+    deathYear: '',
+  });
+  const [retrievedData, setRetrievedData] = React.useState<boolean>(false);
+
   const { ArtifactApplication } = useContractContext();
+  const { Artists } = useContractContext();
   const { accounts } = useWeb3Context();
 
+  const hashToArtist = async (hash: string): Promise<Artist> => {
+    const response = await fetch(hash);
+    return response.json();
+  };
+
   const onSubmit: RegisterOnSubmit = async ({ files, fields }): Promise<void> => {
+    const getArtistInfo = (): void => {
+      if (!fields.artistId && retrievedData) {
+        return;
+      }
+
+      Artists.methods.getArtist(fields.artistId)
+        .call()
+        .then((hash: string) => hashToArtist(hash))
+        .then((artist: Artist) => {
+          setArtist(artist);
+          setRetrievedData(true);
+        })
+        .catch(console.log);
+    };
```

Comment:
I think potentially a better way to do this is to have another field in the ArtifactMetadata (and the FormProvider TextFields) called artistWallet and in `onArtistChange` in ArtistSelection, we can also set the artistWallet. This would remove the need to call the blockchain and then ipfs again (which seems to slow down registering a piece a lot) to get the artist's wallet as we already have this information in ArtistSelection .

Reply:
Will have a look into doing this tomorrow as it would save considerable time when running it